src/runtime_src/xocl/core/kernel.cpp

```diff
@@ -433,7 +433,12 @@ validate_cus(const device* device, unsigned long argidx, int memidx) const
 {
   XOCL_DEBUG(std::cout,"xocl::kernel::validate_cus(",argidx,",",memidx,")\n");
   xclbin::memidx_bitmask_type connections;
-  connections.set(memidx);
+  auto dev = xrt_core::get_userpf_device(device->get_handle());
+  int l_idx = dev->get_groupRange(memidx).first;
+  int h_idx = dev->get_groupRange(memidx).second;
+  for (int i = l_idx; i<h_idx; i++)
+        connections.set(i);
+
   auto end = m_cus.end();
   for (auto itr=m_cus.begin(); itr!=end; ) {
     auto cu = (*itr);
```

Comment:
I don't really want changes in OCL at all.   Today OCL gets the bank index from code that parses mem_topology and connectivity.  It should continue to do so, only it should be using modified mem_topology and connectivity sections stored in the core/common/device class.